liquid-compiler/src/grammar.pest

```diff
@@ -0,0 +1,56 @@
+WHITESPACE = _{" "}
+LINE_OR_SPACE = _{ NEWLINE | WHITESPACE }
+LiquidFile = ${ SOI ~ Element* ~ EOI }
+
+// Element-level parsing
+Element = _{ Expression | Tag | Raw } 
+
+TagStart = _{ (LINE_OR_SPACE* ~ "{%-") | "{%" }
+TagEnd =   _{ ("-%}" ~ LINE_OR_SPACE*) | "%}" }
+ExpressionStart = _{ (LINE_OR_SPACE* ~ "{{-") | "{{" }
+ExpressionEnd =   _{ ("-}}" ~ LINE_OR_SPACE*) | "}}" }
+
+Tag = !{ TagStart ~ Identifier ~ TagToken* ~ TagEnd }
+Expression = !{ ExpressionStart ~ FilterChain ~ ExpressionEnd }
+// Not allowing Tag/Expression Start/End might become a problem 
+// for {% raw %} tag: unclosed delimiters won't be accepted
+Raw = @{ (!(TagStart | ExpressionStart | TagEnd | ExpressionEnd) ~ ANY)+ }              
+
+
+// Inner parsing        
+Identifier = @{ (ASCII_ALPHA | "_")+ }
+
+Variable = { Identifier 
+           ~ ( ("." ~ Identifier)
+             | ("[" ~ Value ~ "]")
+             )* 
+           }
+Value = { Literal | Variable }
```

Comment:
👍  Looks like the grammar is being extended for #145

Reply:
Looks like a `Value` can also be a range?

https://github.com/Shopify/liquid/blob/master/lib/liquid/expression.rb#L28
https://github.com/Shopify/liquid-c/blob/master/ext/liquid_c/parser.c#L131

Reply:
Created https://github.com/cobalt-org/liquid-rust/issues/224 for ranges as values